ambari-agent/src/main/python/ambari_agent/security.py

```diff
@@ -79,8 +83,21 @@ def connect(self):
 
       conn = AmbariStompConnection(self.connection_url, ssl_options=ssl_options)
 
-    self.establish_connection(conn)
-    return conn
+      try:
+        self.establish_connection(conn)
+        logger.info('SSL connection established. Two-way SSL authentication '
+                    'completed successfully.')
+      except ssl.SSLError as err:
+        logger.error('Two-way SSL authentication failed. Ensure that '
+                     'server and agent certificates were signed by the same CA '
+                     'and restart the agent. '
+                     '\nIn order to receive a new agent certificate, remove '
+                     'existing certificate file from keys directory. As a '
+                     'workaround you can turn off two-way SSL authentication in '
+                     'server configuration(ambari.properties) '
+                     '\nExiting..')
+        raise err
```

Comment:
Can we please use "raise" instead of "raise err". This way we can preserve the original stack trace to increase debugibility in this place.

Also we can remove " as err" than.

Reply:
I've made changes as per your request, @aonishuk